service_capacity_modeling/tools/instance_families.py

```diff
@@ -0,0 +1,265 @@
+        "cpu_ipc_scale": ICX_IPC,
+        "cpu_turbo_single_ghz": 3.5,
+        "cpu_turbo_all_ghz": 3.5,
+    },
+    "m7a": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": GENOA_IPC,
+        "cpu_turbo_single_ghz": 3.7,
+        "cpu_turbo_all_ghz": 3.7,
+    },  # is this turbo speed correct?
+    "m7i": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": SPR_IPC,
+        "cpu_turbo_single_ghz": 3.8,
+        "cpu_turbo_all_ghz": 3.2,
+    },
+    # "mac2-m2pro": {'xl_iops': None, 'io_latency_curve': 'ssd', 'cpu_ipc_scale': None},
+    # "p4d": {
+    #     'xl_iops': None,
+    #     'io_latency_curve': '5th-gen-ssd',
+    #     'cpu_ipc_scale': None
+    # },
+    # "p4de": {
+    #     'xl_iops': None,
+    #     'io_latency_curve': '5th-gen-ssd',
+    #     'cpu_ipc_scale': None
+    # },
+    # "p5": {
+    #     'xl_iops': None,
+    #     'io_latency_curve':'6th-gen-ssd',
+    #     'cpu_ipc_scale': None
+    # },
+    # "p5en": {
+    #     'xl_iops': None,
+    #     'io_latency_curve': '6th-gen-ssd',
+    #     'cpu_ipc_scale': None
+    # },
+    "r4": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": HSW_IPC,
+        "cpu_turbo_single_ghz": 3.0,
+        "cpu_turbo_all_ghz": 2.6,
+    },  # all-core is a guess
+    "r5": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": SKX_IPC,
+        "cpu_turbo_single_ghz": 3.5,
+        "cpu_turbo_all_ghz": 3.1,
+    },
+    # exclude r5d and r5dn because they are in the manual list
+    # "r5d": {
+    #     "xl_iops": None,
+    #     "io_latency_curve": "5th-gen-ssd",
+    #     "cpu_ipc_scale": SKX_IPC,
+    #     "cpu_turbo_single_ghz": 3.5,
+    #     "cpu_turbo_all_ghz": 3.1,
+    # },
+    # "r5dn": {
+    #     "xl_iops": None,
+    #     "io_latency_curve": "5th-gen-ssd",
+    #     "cpu_ipc_scale": SKX_IPC,
+    #     "cpu_turbo_single_ghz": 3.5,
+    #     "cpu_turbo_all_ghz": 3.1,
+    # },
+    "r5n": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": SKX_IPC,
+        "cpu_turbo_single_ghz": 3.5,
+        "cpu_turbo_all_ghz": 3.1,
+    },
+    "r6a": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": MILAN_IPC,
+        "cpu_turbo_single_ghz": 3.6,
+        "cpu_turbo_all_ghz": 3.6,
+    },
+    "r6i": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": ICX_IPC,
+        "cpu_turbo_single_ghz": 3.5,
+        "cpu_turbo_all_ghz": 3.5,
+    },
+    "r6id": {
+        "xl_iops": None,
+        "io_latency_curve": "6th-gen-ssd",
+        "cpu_ipc_scale": ICX_IPC,
+        "cpu_turbo_single_ghz": 3.5,
+        "cpu_turbo_all_ghz": 3.5,
+    },
+    "r7a": {
+        "xl_iops": None,
+        "io_latency_curve": None,
+        "cpu_ipc_scale": GENOA_IPC,
+        "cpu_turbo_single_ghz": 3.7,
```

Comment:
How do the turbo affect the calculations or are you just future proofing? Definitely leave it to y'all on if this is how you want to maintain it, definitely agreed this plus auto_shape is probably easier than before, but I am curoius if this needs shape level granularity (one thing I liked about the auto_shape was we more expressed those opinions for a generation/architecture pair rather than per shape).